src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,591 @@
+        # read_request_3 = await self.default_controller.ReadAttribute(
+        #     self.dut_node_id, [(0, Clusters.Objects.Descriptor.Attributes.FeatureMap)], dataVersionFilters=data_version_filter_2)
+
+        # Step 27
+
+        # TH sends a Read Request Message to the DUT to read any supported attribute/wildcard on a particular cluster say A with the DataVersionFilter Field not set.
+        # DUT sends back the attribute value with the DataVersion of the cluster A.
+        # TH sends a Read Request Message to read any supported attribute/wildcard on cluster A and any supported attribute/wildcard on another cluster B.
+        # DataVersionList field should only contain the DataVersion of cluster A.
+
+        # Verify that the DUT sends a report data action with the attribute value from the cluster B to the TH.
+        # Verify that the DUT does not send the attribute value from cluster A.
+        self.print_step(
+            27, "Send the Read Request Message to read any supported attribute/wildcard on a particular cluster say A with the DataVersionFilter Field not set")
+        read_request_27_1_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [server_list_attr])
+
+        data_version_1 = read_request_27_1_1[0][Clusters.Descriptor][Clusters.Attribute.DataVersion]
+
+        # data_version_filter_1 = [(0, Clusters.Descriptor, data_version_1)]
+
+        read_request_27_2_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, Clusters.BasicInformation.Attributes.NodeLabel)])
+        data_version_2 = read_request_27_2_1[0][Clusters.BasicInformation][Clusters.Attribute.DataVersion]
+        # data_version_filter_2 = [(0, Clusters.Descriptor, data_version_2)]
+
+        asserts.assert_not_equal(data_version_1, data_version_2)
+
+        # Step 28
+
+        # TH sends a Read Request Message to the DUT to read something(Attribute) which is larger than 1 MTU(1280 bytes) and per spec can be chunked +
+
+        # Verify on the TH that the DUT sends a chunked data message with the SuppressResponse field set to False for all the messages except the last one.
+        # Verify the last chunked message sent has the SuppressResponse field set to True.
+
+        # self.print_step(28, "Send the Read Request Message to read something(Attribute) which is larger than 1 MTU(1280 bytes) and per spec can be chunked +")
+        # This apparently already exists in TCP tests -- remove?
+
+        # Step 29
+
+        # TH sends a Read Request Message to the DUT to read a non global attribute from all clusters at that Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint, Attribute = Specific Non Global Attribute]] +
+
+        # On the TH verify that the DUT sends an error message and not the value of the attribute.
+        self.print_step(29, "Send the Read Request Message to the DUT to read a non global attribute from all clusters at that Endpoint")
+        found_non_global = False
```

Comment:
Suggest just using something from the descriptor cluster rather than going though the process of selecting something. you'll have to specify the attribute path directly.

Reply:
Hardcoded to use Clusters.Objects.Descriptor.Attributes.ServerList